extension/bootstrap.js

```diff
@@ -1,13 +1,39 @@
+// for shield-study-addon-utils
+/* eslint no-underscore-dangle: "off" */
+
 const { classes: Cc, interfaces: Ci, utils: Cu } = Components;
 Cu.import("resource://gre/modules/Services.jsm");
 Cu.import("resource://gre/modules/Console.jsm");
 Cu.import("resource://gre/modules/AppConstants.jsm");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "studyUtils",
+  "resource://share-button-study/StudyUtils.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "config",
+  "resource://share-button-study/Config.jsm");
+
+const REASONS = {
+  APP_STARTUP: 1,      // The application is starting up.
+  APP_SHUTDOWN: 2,     // The application is shutting down.
+  ADDON_ENABLE: 3,     // The add-on is being enabled.
+  ADDON_DISABLE: 4,    // The add-on is being disabled. (Also sent during uninstallation)
+  ADDON_INSTALL: 5,    // The add-on is being installed.
+  ADDON_UNINSTALL: 6,  // The add-on is being uninstalled.
+  ADDON_UPGRADE: 7,    // The add-on is being upgraded.
+  ADDON_DOWNGRADE: 8,  // The add-on is being downgraded.
+};
+// REASONS[str] = int and REASONS[int] = str
+for (const key in REASONS) {
+  if (Object.prototype.hasOwnProperty.call(REASONS, key)) {
+    REASONS[REASONS[key]] = key;
```

Comment:
I believe this is to be able to have both a map from both:
- reason -> integer value (ie.  `{ APP_STARTUP: 1 }`)
- integer value -> reason (ie. `{ 1: APP_STARTUP }`)

However, I agree that this is not necessary.